spec/turtle-bnf.html

```diff
@@ -1,304 +1,406 @@
-    <td><code>PN_LOCAL</code></td>
-    <td>::=</td>
-    <td>(<a href='#grammar-production-PN_CHARS_U'>PN_CHARS_U</a> <code>| </code> '<code class='grammar-literal'>:</code>' <code>| </code> [<code class='grammar-chars'>0-9</code>] <code>| </code> <a href='#grammar-production-PLX'>PLX</a>) ((<a href='#grammar-production-PN_CHARS'>PN_CHARS</a> <code>| </code> '<code class='grammar-literal'>.</code>' <code>| </code> '<code class='grammar-literal'>:</code>' <code>| </code> <a href='#grammar-production-PLX'>PLX</a>)<code class='grammar-star'>*</code> (<a href='#grammar-production-PN_CHARS'>PN_CHARS</a> <code>| </code> '<code class='grammar-literal'>:</code>' <code>| </code> <a href='#grammar-production-PLX'>PLX</a>))?</td>
-</tr>
-            <tr id="grammar-production-PLX" class='grammar-token'>
-    <td>[169s]</td>
-    <td><code>PLX</code></td>
-    <td>::=</td>
-    <td><a href='#grammar-production-PERCENT'>PERCENT</a> <code>| </code> <a href='#grammar-production-PN_LOCAL_ESC'>PN_LOCAL_ESC</a></td>
-</tr>
-            <tr id="grammar-production-PERCENT" class='grammar-token'>
-    <td>[170s]</td>
-    <td><code>PERCENT</code></td>
-    <td>::=</td>
-    <td>'<code class='grammar-literal'>%</code>' <a href='#grammar-production-HEX'>HEX</a> <a href='#grammar-production-HEX'>HEX</a></td>
-</tr>
-            <tr id="grammar-production-HEX" class='grammar-token'>
-    <td>[171s]</td>
-    <td><code>HEX</code></td>
-    <td>::=</td>
-    <td>[<code class='grammar-chars'>0-9</code>] <code>| </code> [<code class='grammar-chars'>A-F</code>] <code>| </code> [<code class='grammar-chars'>a-f</code>]</td>
-</tr>
-            <tr id="grammar-production-PN_LOCAL_ESC" class='grammar-token'>
-    <td>[172s]</td>
-    <td><code>PN_LOCAL_ESC</code></td>
-    <td>::=</td>
-    <td>'<code class='grammar-literal'>\</code>' ('<code class='grammar-literal'>_</code>' <code>| </code> '<code class='grammar-literal'>~</code>' <code>| </code> '<code class='grammar-literal'>.</code>' <code>| </code> '<code class='grammar-literal'>-</code>' <code>| </code> '<code class='grammar-literal'>!</code>' <code>| </code> '<code class='grammar-literal'>$</code>' <code>| </code> '<code class='grammar-literal'>&amp;</code>' <code>| </code> "<code class="grammar-literal">&#x27;</code>" <code>| </code> '<code class='grammar-literal'>(</code>' <code>| </code> '<code class='grammar-literal'>)</code>' <code>| </code> '<code class='grammar-literal'>*</code>' <code>| </code> '<code class='grammar-literal'>+</code>' <code>| </code> '<code class='grammar-literal'>,</code>' <code>| </code> '<code class='grammar-literal'>;</code>' <code>| </code> '<code class='grammar-literal'>=</code>' <code>| </code> '<code class='grammar-literal'>/</code>' <code>| </code> '<code class='grammar-literal'>?</code>' <code>| </code> '<code class='grammar-literal'>#</code>' <code>| </code> '<code class='grammar-literal'>@</code>' <code>| </code> '<code class='grammar-literal'>%</code>')</td>
-</tr>
+
```

Comment:
Check again; I added classes to the code blocks surrounding grammar tokens and adjusted styling. Also, found a parsing bug which affected `STRING_LITERAL_LONG_QUOTE` and `STRING_LITERAL_LONG_SINGLE_QUOTE`.

Reply:
How can I see it? Neither preview nor local copy do not display the grammar file.

The githack is showing original but looking at the raw HTML and a local hack, somethings have changed for the better but not showing there. Does the URL change? (I've hacked index.html locally for now.)

PN_CHARS_BASE seems to have some table boarders showing ... look like two vbars when the HTML has only one so I guess CSS table boarder.
![image](https://github.com/w3c/rdf-turtle/assets/101362/d1949942-0f95-451b-aa43-4d27ee20be40)

Inconsistent coloring of quotes and used quotes.
![image](https://github.com/w3c/rdf-turtle/assets/101362/a9bbb680-3aa7-49b4-92aa-ffe049157e90)

in 42 inside the grammar-`' '`, the literal  ", "", """ should be orange.

I'd prefer the vbar to be grey. I find the use of orange for single grammar operators unhelpful - reserve that for literal use of the character and keep the grammar-level operator uses as grey (and a bit darker please). It would help with e.g. 

![image](https://github.com/w3c/rdf-turtle/assets/101362/fb6464d0-6a7b-4072-b15b-ebe244188093)

where there are grammar literals and operators.

Reply:
And one of those "shouldn't have looked "things:

The grammar rule 7 (not new - it is the same in RDF 1.1)

>The [162s] ANON ::= '[' WS* ']' token allows any amount of white space and comments between []s. The single space version is used in the grammar for clarity.

doesn't seem to be needed at all. The single space version `[ ]` does not appear in the grammar.

Reply:
This
![image](https://github.com/w3c/rdf-turtle/assets/101362/147faa4d-e2b0-497b-9c03-93df49c6ccf4)
used to be
![image](https://github.com/w3c/rdf-turtle/assets/101362/8e2e0c03-772d-4d2c-9bab-b95f1191510b)

Reply:
> Is there a way to see the outcome online? Neither preview nor local copy display the grammar file.The githack is showing original but somethings have changed for the better but not showing there. Does the URL change?

Sorry, I didn't update the GitHack link; I'll do so after the next update. (Pain that PR Preview isn't working, and wouldn't pick it up in any case).

> `PN_CHARS_BASE` seems to have some table boarders showing ... look like two vbars when the HTML has only one so I guess CSS table boarder. ![image](https://user-images.githubusercontent.com/101362/243065975-d1949942-0f95-451b-aa43-4d27ee20be40.png)

Yeah, that's an odd regression.

> Inconsistent coloring of quotes and used quotes. ![image](https://user-images.githubusercontent.com/101362/243066113-a9bbb680-3aa7-49b4-92aa-ffe049157e90.png)

> in 42 inside the grammar-`' '`, the literal ", "", """ should be orange.

Some literal ranges are missing markup, should be `<code class="grammar-literal">'"""'</code>`, although I think the CSS sets that to grey; may need to tweak CSS classes to get more consistent. Most everything should have its own CSS class.

> And one of those "shouldn't have looked "things:
> 
> The grammar rule 7 (not new - it is the same in RDF 1.1)
> 
> > The [162s] ANON ::= '[' WS* ']' token allows any amount of white space and comments between []s. The single space version is used in the grammar for clarity.
> 
> doesn't seem to be needed at all. The single space version `[ ]` does not appear in the grammar.

+1; remnant from 1.1.


> This
> ![image](https://user-images.githubusercontent.com/101362/243066854-147faa4d-e2b0-497b-9c03-93df49c6ccf4.png)
> used to be
> ![image](https://user-images.githubusercontent.com/101362/243066894-8e2e0c03-772d-4d2c-9bab-b95f1191510b.png)

Using `@terminals` makes it an EBNF declaration, where the other was a heading. I personally prefer that, be we may also tweak the CSS to improve the display.